product-scenarios/2-collaborative-api-development/2.2-manage-visibility-of-apis-in-publisher-portal/2.2.1-prevent-unintended-parties-from-modifying-and-viewing-APIs/src/test/java/org/wso2/am/scenario/tests/rest/api/publisherVisibilityRoleRestriction/publisherAccessControlNegativeTestCase.java

```diff
@@ -0,0 +1,168 @@
+    private static final String TEST_PUBLISHER_USERNAME = "creatorPublisherUser2";
+    private static final String PUBLISHER_ROLE = "publisherTestRole";
+    private static final String PASSWORD = "Wso2123!";
+    private static final String ADMIN_LOGIN_USERNAME = "admin";
+    private static final String ADMIN_LOGIN_PW = "admin";
+    
+    @BeforeClass(alwaysRun = true)
+    public void init() throws APIManagementException {
+
+        String apiPublishPermission = "/permission/admin/manage/api/publish";
+        String apiCreatePermission = "/permission/admin/manage/api/create";
+        String loginPermission = "/permission/admin/login";
+
+        createRole(ADMIN_LOGIN_USERNAME, ADMIN_LOGIN_PW, PUBLISHER_ROLE, new String[]{apiPublishPermission,
+                apiCreatePermission, loginPermission});
+        createUser(CREATOR_PUBLISHER_USERNAME, PASSWORD, new String[]{PUBLISHER_ROLE},
+                ADMIN_LOGIN_USERNAME, ADMIN_LOGIN_PW);
+        usersToDeleteList.add(CREATOR_PUBLISHER_USERNAME);
+        rolesToDeleteList.add(PUBLISHER_ROLE);
+        apiPublisherRestClient = new APIPublisherRestClient(publisherURL);
+    }
+    
+    @Test(description = "2.2.1.1")
+    private void testAccessControlUsingNoneExistingRoles() throws APIManagerIntegrationTestException {
+
+        apiProvider = CREATOR_PUBLISHER_USERNAME;
+        apiName = "testAccessControlUsingNoneExistingRoles";
+        apiContext = "/testAccessControlUsingNoneExistingRoles";
+        apiRole = "somerole";
+        apiVersion = "1.0.0";
+
+        apiPublisherRestClient.login(CREATOR_PUBLISHER_USERNAME, PASSWORD);
+        assertTrue(validateRoles(apiRole).getData().contains("false"));
+        apiPublisherRestClient.logout();
+    }
+
+    @Test(description = "2.2.1.2")
+    private void testVisibilityOfRestrictedApi() throws MalformedURLException, APIManagerIntegrationTestException,
+            APIManagementException {
+
+        apiProvider = CREATOR_PUBLISHER_USERNAME;
+        apiName = "testVisibilityOfRestrictedApi";
+        apiContext = "/testVisibilityOfRestrictedApi";
+        apiVersion = "1.0.0";
+        apiRole = PUBLISHER_ROLE;
+
+        Map<String, String> apiDeleteParams = new HashMap<>();
+        apiDeleteParams.put("apiName", apiName);
+        apiDeleteParams.put("apiProvider", apiProvider);
+        apiDeleteParams.put("apiVersion", apiVersion);
+        apisToDeleteList.add(apiDeleteParams);
+        apiPublisherRestClient.login(CREATOR_PUBLISHER_USERNAME, PASSWORD);
+        verifyResponse(createRestrictedApi(apiPublisherRestClient));
+        apiPublisherRestClient.logout();
+        createUserWithPublisherAndCreatorRole(TEST_PUBLISHER_USERNAME, PASSWORD, ADMIN_LOGIN_USERNAME, ADMIN_LOGIN_PW);
+        usersToDeleteList.add(TEST_PUBLISHER_USERNAME);
+        apiPublisherRestClient.login(TEST_PUBLISHER_USERNAME, PASSWORD);
+        assertFalse(isApiAvailable(apiName, apiPublisherRestClient));
```

Comment:
This api should be visible to admin. Shall we assert that also if there is no separate test for that?